packages/experiments/src/components/FolderCover/FolderCover.tsx

```diff
@@ -73,11 +73,26 @@ export class FolderCover extends React.Component<IFolderCoverProps, IFolderCover
       metadata,
       signal,
       children,
+      isFluent,
       ...divProps
     } = this.props;
 
     const assets = ASSETS[size][type];
+    const MetadataIcon = (): JSX.Element | null => {
```

Comment:
React trivia:
When you define an inline function, and then create a React element around it, it messes up React's diff logic. Reason being that React only compares the 'type', and it's a new function every time!
For this behavior, I suggest not creating functions at all. Instead, just do:
```ts
const metadataIcom = metadata ? (
  <span className={ ... }>
    { metadata }
  </span>
) : null;
...
```
Then just emit as:
```ts
<React.Fragment>
  { metadataIcon }
  { signalIcon }
</React.Fragment>
```

Reply:
Agreed with Thomas there. Also, you're creating new component here, why promote them as real functional components?